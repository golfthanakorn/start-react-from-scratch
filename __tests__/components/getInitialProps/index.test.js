import React from 'react'
import { shallow } from 'enzyme'
import Example from '../../../components/getInitialProps'

describe('snapshot', () => {
  it('should match snapshot', () => {
    const component = shallow(<Example />)

    expect(component).toMatchSnapshot()
  })
})

describe('getInitialProps', () => {
  it('should return data = server', () => {
    const result = Example.getInitialProps({ isServer: true })

    expect(result).toStrictEqual({ data: 'server' })
  })

  it('should return data = client', () => {
    const result = Example.getInitialProps({ isServer: false })

    expect(result).toStrictEqual({ data: 'client' })
  })
})
