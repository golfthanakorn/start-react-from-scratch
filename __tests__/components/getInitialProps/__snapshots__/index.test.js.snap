// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should match snapshot 1`] = `
<div>
  1
</div>
`;
